### PR TITLE
Hazelcast shutdown

### PR DESCRIPTION
**Pull request by rvanheest**
@janvanmansum WIP

**Still to do:**
- [x] make sure the Hazelcast service works correctly on deasy after deployment
- [x] how to deal with invalid JSON?
- ~~incorporate `sender` and `message_version` in the JSON (also with routing of messages to the various versions of a service - Apache Camel???)~~
- write the JSON message interface in `README.md` (will do that in another PR)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -97,6 +101,13 @@
         </dependency>
         <dependency>
             <groupId>com.hazelcast</groupId>
+            <artifactId>hazelcast</artifactId>
+            <version>3.6.4</version>
```
Comment (janvanmansum) [inline pom.xml]: Why not depend on `3.7`?

File: pom.xml [hunk 2]
```diff
@@ -97,6 +101,13 @@
...
+            <classifier>tests</classifier>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.hazelcast</groupId>
             <artifactId>hazelcast-client</artifactId>
             <version>3.6.4</version>
```
Comment (janvanmansum) [inline pom.xml]: Why not depend on `3.7`?

**Review by janvanmansum — COMMENTED** [review]
Some questions.

File: pom.xml (hunk 1 above)
Comment (rvanheest) [reply pom.xml]: We probably should... I'll add this to my TODO list for today 😉

**Comment by rvanheest** [comment]
I think I got it working on deasy now...
During testing with deasy I discovered some weird deserialization errors with the UUID, so I had to convert them to `String`s as the key of the result map. Unfortunately I wasn't able to replicate this behavior in a unit test inside the project...
Other than that I deployed and run both DOI and URN requests on deasy using both the REST and Hazelcast services.

To monitor Hazelcast, I wrote a little REPL which is not included in this PR but which could serve as a basis of a monitor/interaction tool. LMKWYT

**Pull request merged**